src/converters/cover.ts

```diff
@@ -251,15 +289,44 @@ class CoverHandler implements ServiceHandler {
       this.current_min,
       this.current_max
     );
-
-    this.service.updateCharacteristic(hap.Characteristic.CurrentPosition, characteristicValue);
-
-    if (isStopped) {
-      // Update target position and position state
-      // This should improve the UX in the Home.app
-      this.accessory.log.debug(`${this.accessory.displayName}: cover: assume movement stopped`);
-      this.service.updateCharacteristic(hap.Characteristic.PositionState, hap.Characteristic.PositionState.STOPPED);
-      this.service.updateCharacteristic(hap.Characteristic.TargetPosition, characteristicValue);
+    if (this.motorStateExpose === undefined) {
+      this.service.updateCharacteristic(hap.Characteristic.CurrentPosition, characteristicValue);
+      if (isStopped) {
+        // Update target position and position state
+        // This should improve the UX in the Home.app
+        this.accessory.log.debug(`${this.accessory.displayName}: cover: assume movement stopped`);
+        this.service.updateCharacteristic(hap.Characteristic.PositionState, hap.Characteristic.PositionState.STOPPED);
+        this.service.updateCharacteristic(hap.Characteristic.TargetPosition, characteristicValue);
+      }
+    } else if (this.motorState !== this.motorStatePrevious) {
+      let newPositionState: number;
+      let newTargetPosition: number;
+      switch (this.motorState) {
+        case CoverHandler.MOTOR_STATE_CLOSING:
+          newPositionState = hap.Characteristic.PositionState.DECREASING;
+          newTargetPosition = 0;
```

Comment:
@burmistrzak Just wondering, did you also check the UX by setting a target position from the app to a value less than 100 or greater than 0?

I'm curious if these fixed values of `0` and `100` for `newTargetPosition` are okay, or if they may also lead to weird behavior.

Reply:
I'll merge it for now. We can always improve it later, if needed.